Examples/MAX32660/SPIMSS_DMA/main.c

```diff
@@ -0,0 +1,216 @@
+#if SPIMSS_AUTO_DMA_HANDLING
+        dma_tx_completed = 1;
+#else
+        //Disable interrupts.
+        dma_tx_completed = 1;
+        MXC_DMA_DisableInt(txChannel);
+        MXC_DMA_SetChannelInterruptEn(txChannel, false, false); // Disable ctz interrupt.
+        MXC_DMA_ReleaseChannel(txChannel);
+        MXC_SPIMSS_SetTXDMAChannel(SPIMSS, -1);
+        MXC_SPIMSS->dma &= ~MXC_F_SPIMSS_DMA_TX_DMA_EN; // Disable SPIMSS TX DMA requests
+
+        if (req->deass) {
+            MXC_SPIMSS->mode |= MXC_F_SPIMSS_MODE_SSV; // Set Slave Select to HIGH.
+        }
+        MXC_SPIMSS->ctrl &= ~MXC_F_SPIMSS_CTRL_ENABLE; // Disable SPIMSS transaction.
+#endif
+    }
+}
+
+int main()
+{
+    int i = 0;
+
+    printf("\n\n************** SPIMSS-DMA Master Demo ****************\n");
+    printf("This example configures the SPIMSS to send data between the MISO (P0.10) and\n");
+    printf("MOSI (P0.11) pins over dma channel 0 and dma channel 1.  Connect these two pins\n");
+    printf("together. This demo shows 1024 byte data transfer for 100 times using dma.\n");
+    printf("During this demo you may see junk data printed to the serial port because the\n");
+    printf("console UART shares the same pins as the SPIMSS.One DMA channel is used as tx from\n");
+    printf("memory to SPIMSS TX FIFO and another dma channel 1 is used for reading data from\n");
+    printf("SPIMSS RX FIFO to memory.\n\n");
+
+    // Preparation of the transmission data and receive data buffer.
+    for (i = 0; i < NUMBER_OF_UNIT_SIZE; i++) {
+        tx_data[i] = DATA_VALUE + i;
+        rx_data[i] = 0;
+    }
+
+    // Initialize SPIMSS with MAP_A.
+    // For MAP_A P0.10:MISO, P0.11:MOSI, P0.12:SCK, P0.13:SSEL
+    if (0 != MXC_SPIMSS_Init(SPIMSS, 0, SPI_SPEED, MAP_A)) {
+        return -1;
+    }
+
+    // Prepare the SPIMSS data transaction request.
+    spimss_dma_req.tx_data = (void *)tx_data;
+    spimss_dma_req.rx_data = (void *)rx_data;
+    spimss_dma_req.len = NUMBER_OF_UNIT_SIZE;
+    spimss_dma_req.bits = 16; // This must be >= 8
+    spimss_dma_req.deass = 1;
+    spimss_dma_req.tx_num = 0;
+    spimss_dma_req.rx_num = 0;
+    spimss_dma_req.callback = spimss_callback;
+
+    for (i = 0; i < TEST_COUNT; i++) {
+// Setting auto dma handlers enable acquiring the tx and rx channels
+// automatically in the background.
+#if SPIMSS_AUTO_DMA_HANDLING
+        MXC_SPIMSS_SetAutoDMAHandlers((SPIMSS), true);
+#else
+        MXC_DMA_Init();
+        txChannel = MXC_DMA_AcquireChannel();
+        rxChannel = MXC_DMA_AcquireChannel();
+        if (0 > txChannel || 0 > rxChannel) {
+            break;
+        }
+        MXC_SPIMSS_SetTXDMAChannel(SPIMSS, txChannel);
+        MXC_SPIMSS_SetRXDMAChannel(SPIMSS, rxChannel);
+#endif
+
+        // Calling SPIM transaction function for communication over DMA.
+        if (E_NO_ERROR != MXC_SPIMSS_MasterTransDMA(SPIMSS, &spimss_dma_req)) {
+            return -1;
```

Comment:
Similar to above - return the error that the function returns instead of -1.

Reply:
Updated